pdr_backend/pdr_dashboard/util/db.py

```diff
@@ -74,16 +78,12 @@ def _query_db(self, query: str, scalar=False) -> Union[List[dict], Exception]:
             dict: Query result.
         """
         try:
-            db = DuckDBDataStore(self.lake_dir, read_only=True)
-
+            # If scalar, fetch a single result
             if scalar:
-                result = db.query_scalar(query)
-            else:
-                df = db.query_data(query)
-                result = df.to_pandas()
-
-            db.duckdb_conn.close()
-            return result
+                result = duckdb.execute(query).fetchone()
+                return result[0] if result and len(result) == 1 else result
+            df = duckdb.execute(query).pl()
```

Comment:
what is the purpose of pl?

Reply:
to convert the result to Polars DataFrame

Reply:
why? Dash Table only accepts pandas. And the next line is literally a conversion to pandas which worked without the need for pl(). So instead of converting it to pl and then pandas, why not have it directly as pandas?